garage/tf/core/parameter.py

```diff
@@ -37,7 +36,9 @@ def parameter(input_var,
             initializer=initializer,
             trainable=trainable)
 
-        broadcast_shape = tf.concat(
-            axis=0, values=[tf.shape(input_var)[:-1], [length]])
-        p_broadcast = broadcast_to(p, shape=broadcast_shape)
-        return p_broadcast
+        ndim = input_var.get_shape().ndims
+        reshaped_p = tf.reshape(p, (1, ) * (ndim - 1) + (length, ))
+        tile_arg = tf.concat(
```

Comment:
why tile instead of broadcast?

I think broadcast is the canonical TensorFlow API to use. It allows the graph compiler to resolve the shape and optimize. Tiling forces a particular memory layout which might be inefficient.

In any case, this is not in scope for the PR unless it's necessary for models to work.

Reply:
It seems like these should become keras Layers at some point